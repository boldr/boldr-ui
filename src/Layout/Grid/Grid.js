/* @flow */
import * as React from 'react';
import classNames from 'classnames';

export type Props = {
  fluid: boolean,
  className: ?string,
  style: ?Object,
  children: Array<React.Node>,
  componentClass: string,
};

class Grid extends React.PureComponent<Props, *> {
  static defaultProps = {
    fluid: false,
    componentClass: 'div',
  };
  props: Props;
  render() {
    const ComponentClass = this.props.componentClass;

    const classes = classNames(
      {
        grid: !this.props.fluid,
        'grid--fluid': this.props.fluid,
      },
      this.props.className,
    );

    return (
      <ComponentClass className={classes} style={this.props.style}>
        {this.props.children}
      </ComponentClass>
    );
  }
}

export default Grid;
