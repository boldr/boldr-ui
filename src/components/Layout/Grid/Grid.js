import React from 'react';
import classNames from 'classnames';

export type Props = {
  fluid?: boolean,
  className?: string,
  style?: Object,
  children?: number | string | React.Element | Array<any>,
  componentClass: ?string,
};

const Grid = ({ fluid, className, style, children, componentClass }: Props) => {
  const ComponentClass = componentClass;

  const classes = classNames(
    {
      grid: !fluid,
      grid__fluid: fluid,
    },
    className,
  );

  return (
    <ComponentClass className={classes} style={style}>
      {children}
    </ComponentClass>
  );
};

Grid.defaultProps = {
  componentClass: 'div',
};

export default Grid;
