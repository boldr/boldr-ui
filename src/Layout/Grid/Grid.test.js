import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Grid from './Grid';

describe('<Grid>', () => {
  it('should render "contained"', () => {
    const wrapper = mount(<Grid />);
    const wrapperInner = wrapper.find('.grid');
    expect(wrapperInner.is('.grid')).toBe(true);
  });
  it('should render with fluid props', () => {
    const wrapper = mount(<Grid fluid />);
    const wrapperInner = wrapper.find('.grid--fluid');
    expect(wrapperInner.is('.grid--fluid')).toBe(true);
  });
  it('should render with the passed html tag', () => {
    const wrapper = mount(<Grid componentClass="section" />);
    const wrapperInner = wrapper.find('section');
    expect(wrapperInner.is('.grid')).toBe(true);
  });
  it('should match the snapshot', () => {
    const wrapper = shallow(<Grid />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
