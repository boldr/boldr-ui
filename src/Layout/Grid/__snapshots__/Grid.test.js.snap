// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid> should match the snapshot 1`] = `
<div
  className="grid"
/>
`;
